### PR TITLE
`CI`: added `iOS 17` job

### PR DESCRIPTION
**Pull request by NachoSoto**
https://discuss.circleci.com/t/xcode-15-beta-1-released/48298

### Depends on :
- #2603
- #2602
- #2605
- #2606
- #2607

### CODE REVIEW

**Comment by NachoSoto** [comment]
I'd suggest merging that even though we still have some failures, so we can keep an eye until it's all resolved (I have a bunch of open PRs that solve them).

File: .circleci/config.yml [hunk 1]
```diff
@@ -315,6 +315,32 @@ jobs:
...
+          condition: << pipeline.parameters.generate_snapshots >>
+          steps:
+            - run:
+                name: Run create_snapshot_pr 
+                command: bundle exec fastlane create_snapshot_pr version:"ios-17"
+      - store_test_results:
+          path: fastlane/test_output
+      - store_artifacts:
+          path: fastlane/test_output/xctest
+          destination: scan-test-output
```
Comment (tonidero) [inline .circleci/config.yml:342]: Should we extract this to be a command? Seems like it's almost repeated (with some small differences) for the other iOS versions... We can do that in a different PR though

**Review by tonidero — APPROVED** [review]
I'm a bit concerned about having tests not passing... even if it's to monitor them... Should we expect failures of them instead so we notice when they start passing? Though please ignore if tests are already passing with other PRs!

**Comment by NachoSoto** [comment]
> Should we expect failures of them instead so we notice when they start passing

That's a good idea, is that a CircleCI thing?
